plugins/hashsum_download/server/__init__.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+###############################################################################
+#  Copyright Kitware Inc.
+#
+#  Licensed under the Apache License, Version 2.0 ( the "License" );
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+###############################################################################
+
+from girder.api import access
+from girder.api.describe import describeRoute, Description
+from girder.api.rest import RestException
+from girder.api.v1.file import File
+
+
+class HashedFile(File):
+
+    SupportedAlgorithms = [
+        'sha512',
+    ]
+
+    def __init__(self, apiRoot):
+        super(File, self).__init__()
+
+        self.resourceName = 'file'
+        apiRoot.file.route('GET', ('hashsum', ':algo', ':hash', 'download'),
```

Comment:
Should we also add `GET /file/hashum/:algo/:hash` to retrieve the file document? It's not necessary for our immediate needs, but should be easy to add.